test/api/students_api_test.rb

```diff
@@ -11,32 +11,48 @@ def app
   end
 
   def test_get_students_with_authentication
-    # Build admin user
-    adminUser = FactoryGirl.build(:user, :admin)
+
+    # Create unit
+    newUnit = FactoryGirl.create(:unit)
+
+    #Create campus
+    newCampus = FactoryGirl.create(:campus)
+
+    # Create student
+    studentUser = FactoryGirl.create(:user, :student)
+
+    # Assign student to the unit
+    newUnit.enrol_student(studentUser, newCampus)
 
     # The get that we will be testing.
-    get with_auth_token '/api/students/?unit_id=1',adminUser
-    response = last_response_body
+    get with_auth_token "/api/students/?unit_id=#{newUnit.id}"
```

Comment:
You need to specify the user making this request. By default it is the first user - but we now want to make that explicit (and not use the first user). Then you are explicit in what kind of user is performing this test.